src/main/java/com/customerApi/api/services/CustomerService.java

```diff
@@ -0,0 +1,150 @@
+package com.customerApi.api.services;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.hibernate.exception.JDBCConnectionException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import com.customerApi.api.exception.CustomerNotFoundException;
+import com.customerApi.api.modals.Customer;
+import com.customerApi.api.repositories.CustomerRepository;
+
+@Service
+public class CustomerService {
+
+	private static final Logger logger = LoggerFactory.getLogger(CustomerService.class);
+
+	@Autowired
+	private CustomerRepository customerRepo;
+
+	public Optional<Integer> addCustomerData(Customer customer) {
+
+		Optional<Integer> customerId = null;
+
+		try {
+
+			Customer cust = customerRepo.save(customer);
+
+			customerId = Optional.of(cust.getId());
+
+		} catch (JDBCConnectionException e) {
+			logger.error("Database connection error" + e);
```

Comment:
In case of this error why you are just printing the error and let the flow move forward, you need to throw exception here right away and pass a meaning message to the front end. Otherwise if you go with this logic you will throw `throw new CustomerNotFoundException(" Customer id-" + customerId);` But as per this exception your database is down, so you should throw different HttpStatus and meaningful message.